deltacat/compute/compactor/steps/dedupe.py

```diff
@@ -363,8 +362,9 @@ def dedupe(
         src_file_records_obj_refs.append(object_ref)
         mat_bucket_to_dd_idx_obj_id[mat_bucket] = (
             dedupe_task_index,
-            cloudpickle.dumps(object_ref),
+            object_ref,
         )
+        del object_ref
```

Comment:
This and our other manual object ref deletes will look confusing to future maintainers who don't have all the context that preceded this PR. Can you include a comment stating why we're including the delete here, what it is meant to help with, and/or a link to an existing discussion about the same on GitHub or the Ray Discussion forums?

Reply:
Resolving this conversation as I'm no longer using the `del` for manual object deletion in my latest commit. Solely removing the `cloudpickle` API calls allows the object memory to free up properly after the job run finishes.